### PR TITLE
add dist:css step to prod-ify CSS file

### LINKED ISSUE #612: Add minified bundles to distribution
Follow-up of #303, #588. Should not include sourcemaps.

### LINKED ISSUE #2122: user-select needs vendor prefixing
There are several uses of the `user-select` property which do not take effect without vendor prefixing in certain browsers; for example, sliders in Firefox after dragging the handle:

slider

 According to [this compatibility chart from MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/user-select#Browser_compatibility) no browser other than Chrome supports the unprefixed version yet. Adding `-webkit-user-select` should satisfy most cases.

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- a few fixes in node-build-scripts:
  - add output option to `sass-compile`
  - add more main fields to `assert-package-layout`
  - add auto-fix string for `file-header` rule in tslint-config
- `sass-dist` script runs postcss with autoprefixer and discard-comments plugins
- add important comment at the top of each scss entry file (preserved in dist output)
- add `.browserslistrc` file instead of individual autoprefixer config

#### Reviewers should focus on:

```
blueprint-datetime.css   185K => 18K
blueprint-icons.css      856B => 988B *
blueprint-labs.css        47B => 162B *
blueprint-select.css      54K => 4.6K
blueprint-timezone.css    57K => 323B
blueprint.css            1.4M => 432K **
docs-theme.css           233K => 29K
table.css                313K => 42K

* - file size increase due to inline sourcemap
** - 70% filesize reduction!!!
```

### CODE REVIEW

**Comment by reiv** [comment]
This would fix #2122 as well. :sparkles:

File: .browserslistrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Browsers that we support
+
+defaults
```
Comment (adidahiya) [inline .browserslistrc]: can we be more explicit? I don't really want to have to look at the documentation for the default. also more comments in this file would be really helpful, to explain its context

File: packages/node-build-scripts/sass-dist [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+# Usage: sass-compile [outdir=lib/css/]
```
Comment (adidahiya) [inline packages/node-build-scripts/sass-dist]: isn't this more like `css-dist`? it has nothing to do with sass.

**Comment by giladgray** [comment]
yarn cache issue with renamed bin script causing build fails

File: .browserslistrc [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# Every autoprefixer run in this repo will use this file for config.
+# Test queries at http://browserl.ist/
+
+# browsers with some usage plus recency:
+> 0.5%
+last 2 versions
+# ensure these are supported:
+Firefox ESR
+IE 11
+# because no one likes a dead browser:
```
Comment (adidahiya) [inline .browserslistrc]: lol

**Comment by adidahiya** [comment]
that's annoying... let's merge into develop and fix things there if necessary

**Pull request merged**